src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1149,15 +1177,860 @@ bool PhaseIterGVN::verify_node_value(Node* n) {
+    //   8423  ConvI2L  === _ 3519  [[ 8424 ]]  #long:-2
+    //   8422  ConvI2L  === _ 8399  [[ 8424 ]]  #long:3..256:www
+    //   8424  AddL  === _ 8422 8423  [[ 8383 ]]  !orig=[8382]
+    //   8016  ConL  === 0  [[ 8383 ]]  #long:0
+    //   8383  SubL  === _ 8016 8424  [[ 8156 ]]  !orig=[8154]
+    //
+    // And then in verification:
+    //   8338  ConL  === 0  [[ 8339 8424 ]]  #long:-2     <----- Was constant folded.
+    //   8422  ConvI2L  === _ 8399  [[ 8424 ]]  #long:3..256:www
+    //   8424  AddL  === _ 8422 8338  [[ 8383 ]]  !orig=[8382]
+    //   8016  ConL  === 0  [[ 8383 ]]  #long:0
+    //   8383  SubL  === _ 8016 8424  [[ 8156 ]]  !orig=[8154]
+    //
+    // So the form changed from:
+    //   c1 - (y + [8423  ConvI2L])
+    // to
+    //   c1 - (y + -2)
+    // but the SubL was not added to the IGVN worklist. Investigate why.
+    // There could be other issues too.
+    //
+    // There seems to be a related AddL IGVN optimization that triggers
+    // the same SubL optimization, so investigate that too.
+    //
+    // Found with:
+    //   java -XX:VerifyIterativeGVN=0100 -Xcomp --version
+    case Op_SubL:
+      return false;
+
+    // SubINode::Ideal does
+    // Convert "x - (y+c0)" into "(x-y) - c0" AND
+    // Convert "c1 - (y+c0)" into "(c1-c0) - y"
+    //
+    // Investigate why this does not yet happen during IGVN.
+    //
+    // Found with:
+    //   test/hotspot/jtreg/compiler/c2/IVTest.java
+    //   -XX:VerifyIterativeGVN=1110
+    case Op_SubI:
+      return false;
+
+    // AddNode::IdealIL does transform like:
+    //   Convert x + (con - y) into "(x - y) + con"
+    //
+    // In IGVN before verification:
+    //   8382  ConvI2L
+    //   8381  ConvI2L  === _ 791  [[ 8383 ]]  #long:0
+    //   8383  SubL  === _ 8381 8382
+    //   8168  ConvI2L
+    //   8156  AddL  === _ 8168 8383  [[ 8158 ]]
+    //
+    // And then in verification:
+    //   8424  AddL
+    //   8016  ConL  === 0  [[ 8383 ]]  #long:0  <--- Was constant folded.
+    //   8383  SubL  === _ 8016 8424
+    //   8168  ConvI2L
+    //   8156  AddL  === _ 8168 8383  [[ 8158 ]]
+    //
+    // So the form changed from:
+    //   x + (ConvI2L(0) - [8382  ConvI2L])
+    // to
+    //   x + (0 - [8424  AddL])
+    // but the AddL was not added to the IGVN worklist. Investigate why.
+    // There could be other issues too. For example with "commute", see above.
+    //
+    // Found with:
+    //   java -XX:VerifyIterativeGVN=0100 -Xcomp --version
+    case Op_AddL:
+      return false;
+
+    // SubTypeCheckNode::Ideal calls SubTypeCheckNode::verify_helper, which does
+    //   Node* cmp = phase->transform(new CmpPNode(subklass, in(SuperKlass)));
+    //   record_for_cleanup(cmp, phase);
+    // This has the effect that these new nodes end up on the IGVN worklist,
+    // but if we now leave verification and IGVN itself, we have nodes on the
+    // worklist, and that should not be (there are asserts against this).
```

Comment:
Sounds like we just need some exception when calling `Ideal` on a `SubTypeCheck` that we can have certain nodes still on the worklist like a `cmp`. Maybe we should add this as a suggestion to the comment?
```suggestion
    // but if we now leave verification and IGVN itself, we have nodes other
    // than 'n' still on the worklist. This will fail with an assert in
    // verify_empty_worklist(). Maybe we just need to add an exception and
    // check that only certain nodes like 'cmp' are still on the worklist. After
    // this check, we can clear the worklist such that verify_empty_worklist() 
    // succeeds.
```